src/components/NewTodoField/NewTodoField.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  FormEvent, useContext, useEffect, useRef, useState,
+} from 'react';
+import { postTodos } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { AuthContext } from '../Auth/AuthContext';
+
+type Props = {
+  onAdd: (todo: Todo) => void;
+  setErrorMessage: (error: string) => void;
+  setVisibleLoader: (loader: boolean) => void;
+  visibleLoader: boolean;
+};
+
+export const NewTodoField: React.FC<Props> = ({
+  onAdd,
+  setErrorMessage,
+  setVisibleLoader,
+  visibleLoader,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const user = useContext(AuthContext);
+
+  const [title, setTitle] = useState<string>('');
+  const [completed, setCompleted] = useState<boolean>(false);
+  const [id, setId] = useState<number>(0);
+
+  const reset = () => {
+    setTitle('');
+    setCompleted(false);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (title) {
+      setVisibleLoader(true);
+
+      postTodos({
+        userId: user?.id || 0,
+        title,
+        completed,
+      }).then(() => {
+        setVisibleLoader(false);
+      }).catch(() => setErrorMessage('Unable to add a todo'));
+
+      onAdd({
+        id,
+        userId: user?.id || 0,
+        title,
+        completed,
+      });
+
+      // eslint-disable-next-line no-param-reassign
+      setId(prevId => prevId + 1);
```

Comment:
you don't need to set any id!

Reply:
Oh, sorry, I knew it just in the last task. in Qna with you:)